Snakefile

```diff
@@ -1143,32 +1162,52 @@ rule encode_sentencepiece_ids:
             "zcat {data_file} | "
             "awk 'BEGIN {{srand(3)}} rand() < {FRACTION_OF_DATA_TO_SPM_ENCODE}' | "
             "spm_encode --model={model_file} --output_format=id | "
-            "python scripts/split_sentencepiece_ids.py"
+            "python scripts/slice_sentencepiece_ids.py"
             " --min-chunk-length {MIN_SPM_CHUNK_LEN}"
             " --max-chunk-length {MAX_SPM_CHUNK_LEN}"
             " --max-length {MAX_SPM_ENCODE_IDS}"
             " --file-prefix {output_prefix}"
         )
 
 
-# Convert the outputs of encode_sentencepiece_ids into .npy files.
-rule numberize_one_sentencepiece_id_file:
+# Split the encoded SentencePiece id's into training, validation, and testing sets.
+rule split_one_sentencepiece_id_file:
     input:
         "data/sentencepiece/{lang}.spm_txt_ids/length_{n}.txt"
     output:
-        "data/sentencepiece/{lang}.spm_ids/length_{n}.npy"
-    shell:
-        "scripts/sentencepiece_id_converter.py {input} {output}"
+        temp("data/sentencepiece/{lang}.spm_txt_ids_training/length_{n}.txt"),
+        temp("data/sentencepiece/{lang}.spm_txt_ids_validation/length_{n}.txt"),
+        temp("data/sentencepiece/{lang}.spm_txt_ids_testing/length_{n}.txt")
+    run:
+        training_file, validation_file, testing_file = output
+        frac_train = FRACTION_OF_ENCODED_SPM_DATA_FOR_TRAINING
+        frac_valid = FRACTION_OF_REMAINING_SPM_DATA_FOR_VALIDATION
+        # touch the outputs so that they are created even if empty
+        shell(
+            "cat {input} | "
+            "python scripts/split_sentencepiece_id_file.py"
+            "  --fraction {frac_train}"
```

Comment:
Is there a reason to run the script twice, instead of having the script take in a testing fraction and a validation fraction?

Reply:
It just seemed simpler to me to write one tool and use it twice.  But I can see the argument for making the two fractions in the snakefile more obvious, so I've changed the script along the lines you propose.